java/CONTRIBUTING.md

```diff
@@ -2,25 +2,48 @@
 > The higher level [Development Guide](https://docs.feast.dev/v/master/project/development-guide)
 > gives contributing to Feast codebase as a whole.
 
-### Overview
+## Overview
 This guide is targeted at developers looking to contribute to Feast components in
 the feast-java Repository:
 - [Feast Serving](#feast-serving)
-- [Feast Java Client](#feast-java-client)
+- [Feast Serving Client](#feast-serving-client)
 
 > Don't see the Feast component that you want to contribute to here?  
 > Check out the [Development Guide](https://docs.feast.dev/v/master/project/development-guide)
 > to learn how Feast components are distributed over multiple repositories.
 
-#### Common Setup
+### Repository structure
+There are four key top level packages:
+- `serving`: Feast Serving (a gRPC service to serve features)
+- `serving-client`: Feast Serving Client (a thin Java client to communicate with Feast serving via gRPC )
```

Comment:
If you want to refactor this to be usable as a fat client, should the `serving` component be split into something like `retriever` which just talks to the online store and `serving` which depends on `retriever` and exposes a gRPC service?

Reply:
yes. I wanted to do that as well, but seemed to make this PR too big. It would be essentially abstracting out most of what lies under ServingServiceV2 into a `FeatureStore` class